# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: media/webrtc/trunk/webrtc/modules/desktop_capture/linux/mouse_cursor_monitor_x11.cc
# Commit: ba49ec6ff971
# Full Hash: ba49ec6ff97150abf58c38c464be7947ef9c591b
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2019-05-29 09:48:47
# Description:
#   Bug 1545247 - Fix crash in MouseCursorMonitorX11::CaptureCursor() on linux32; r=ng
#   
#   This is a cherry pick of upstream commit c1187ab34bdf836bd33f7f050d525184eba4cd20.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D32826
# ==============================================================================

diff -r 887a347cb3ef -r ba49ec6ff971 media/webrtc/trunk/webrtc/modules/desktop_capture/linux/mouse_cursor_monitor_x11.cc
--- a/media/webrtc/trunk/webrtc/modules/desktop_capture/linux/mouse_cursor_monitor_x11.cc	Wed May 22 22:47:29 2019 +0000
+++ b/media/webrtc/trunk/webrtc/modules/desktop_capture/linux/mouse_cursor_monitor_x11.cc	Tue May 28 16:03:19 2019 +0000
@@ -211,11 +211,12 @@
    std::unique_ptr<DesktopFrame> image(
        new BasicDesktopFrame(DesktopSize(img->width, img->height)));
 
-  uint64_t* src = reinterpret_cast<uint64_t*>(img->pixels);
+  // Xlib stores 32-bit data in longs, even if longs are 64-bits long.
+  unsigned long* src = img->pixels;
   uint32_t* dst = reinterpret_cast<uint32_t*>(image->data());
   uint32_t* dst_end = dst + (img->width * img->height);
   while (dst < dst_end) {
-    *dst++ = *src++;
+    *dst++ = static_cast<uint32_t>(*src++);
   }
 
   DesktopVector hotspot(std::min(img->width, img->xhot),
